modules/library/src/main/scala/zio/elasticsearch/AggregationResponse.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.json._
+import zio.json.ast.Json.Obj
+import zio.json.ast._
+import zio.json.internal.RetractReader
+
+sealed trait ElasticAggregationResponse
+
+final case class BucketAggregationResponse(
+  @jsonField("doc_count_error_upper_bound")
+  docErrorCount: Int,
+  @jsonField("sum_other_doc_count")
+  sumOtherDocCount: Int,
+  buckets: List[TermsAggregationBucket]
+) extends ElasticAggregationResponse
+
+object BucketAggregationResponse {
+  implicit val decoder: JsonDecoder[BucketAggregationResponse] = DeriveJsonDecoder.gen[BucketAggregationResponse]
+}
+
+sealed trait AggregationBucket
+
+final case class TermsAggregationBucket(
+  key: String,
+  @jsonField("doc_count")
+  docCount: Int,
+  subAggregations: Option[List[BucketAggregationResponse]] = None
+) extends AggregationBucket
+
+object TermsAggregationBucket {
+  implicit val decoder: JsonDecoder[TermsAggregationBucket] = (trace: List[JsonError], in: RetractReader) => {
+    val json = JsonDecoder[Json].unsafeDecode(trace, in)
+
+    json match {
+      case Obj(fields) =>
+        val dynamicFields = fields.flatMap { case (field, data) =>
+          field match {
+            case "key" =>
+              Some(field -> data.toString.replaceAll("\"", ""))
+            case "doc_count" =>
+              data.as[Int].toOption.map(field -> _)
+            case _ =>
+              val objFields = Obj.decoder.decodeJson(data.toString).toOption.get.fields.toMap
+              Some(
+                field -> BucketAggregationResponse(
+                  docErrorCount = objFields("doc_count_error_upper_bound").as[Int].toOption.get,
+                  sumOtherDocCount = objFields("sum_other_doc_count").as[Int].toOption.get,
+                  buckets = objFields("buckets")
+                    .as[List[Json]]
+                    .toOption
+                    .get
+                    .map(bucket => TermsAggregationBucket.decoder.decodeJson(bucket.toJson).toOption)
+                    .map(_.get)
+                )
+              )
+          }
+        }.toMap
+
+        val key      = dynamicFields.get("key").asInstanceOf[Option[String]].get
+        val docCount = dynamicFields.get("doc_count").asInstanceOf[Option[Int]].get
```

Comment:
Why do we have `asInstanceOf[Option[Int]]` and then `.get`? 

Can we replace this with `dynamicFields("doc_count").asInstaceOf[Int]`?